File: client-app/src/models/department.ts


export interface Department {
    id: string
    name: string
    createdTime: string
    updatedTime: string
    employeeIds: string[]
    positionIds: string[]
    employees: string[]
    positions: string[]
  }

export interface DepartmentCreate {
  name: string
  positionIds: string[]
}

export interface DepartmentUpdate {
  id: string
  name: string
  positionIds: string[]
}

File: EmployeeManagement.Api/Entities/Department.cs
namespace EmployeeManagement.Api;

public class Department
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public DateOnly CreatedTime { get; set; }
    public DateOnly UpdatedTime { get; set; }
    public ICollection<Employee> Employee { get; set; } = [];
    public List<DepartmentPosition>? DepartmentPosition { get; set; }

}
